src/copy.jl

```diff
@@ -0,0 +1,172 @@
+#  Copyright 2017, Iain Dunning, Joey Huchette, Miles Lubin, and contributors
+#  This Source Code Form is subject to the terms of the Mozilla Public
+#  License, v. 2.0. If a copy of the MPL was not distributed with this
+#  file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+"""
+    copy_extension_data(data, new_model::AbstractModel, model::AbstractModel)
+
+Return a copy of the extension data `data` of the model `model` to the extension
+data of the new model `new_model`. A method should be added for any JuMP
+extension storing data in the `ext` field.
+"""
+function copy_extension_data end
+
+"""
+    copy_variablewise_constraints(dest::Dict{MOIVAR,
+                                             MOICON{MOI.SingleVariable, S}},
+                                  src::Dict{MOIVAR,
+                                            MOICON{MOI.SingleVariable, S}},
+                                  index_map) where S
+
+Copy the variablewise constraint indices of `src` into `dest` mapping variable
+and constraint indices using `index_map`.
+"""
+function copy_variablewise_constraints(dest::Dict{MOIVAR,
```

Comment:
`copy_single_variable_constraints`? "variablewise" isn't a concept in MOI anymore.